packages/shopify-cart/docs/custom-fragments.md

```diff
@@ -0,0 +1,128 @@
+
+Please take care to ensure that your fragments use the expected GraphQL types. If a fragment is provided with an unexpected GraphQL type, `@nacelle/shopify-cart` will throw an error when the `cartClient` is initialized.
+
+The fragment name (example: `SomeFragment` in `fragment SomeFragment on GraphqlTypeOfInterest`) can be a name of your choosing.
+
+## Examples
+
+### Querying `sellingPlanAllocation` data on cart line items
+
+Before attempting to query Selling Plan data, please ensure that the [`unauthenticated_read_selling_plans` access scope][shopify-access-scopes] is enabled on your Shopify Storefront API token.
+
+The `CartLine` data returned by `@nacelle/shopify-cart` doesn't include `sellingPlanAllocation` data by default. We can query for `sellingPlanAllocation` data by supplying an `EXTEND_CART_LINE` fragment to the client's `customFragments`.
+
+```js
+const cartClient = createShopifyCartClient({
+  shopifyShopId: '<your-shop-id>',
+  shopifyStorefrontAccessToken: '<your-shopify-storefront-api-access-token>',
+  customFragments: {
+    EXTEND_CART_LINE: `
+      fragment CartLine_sellingPlan on CartLine {
+        sellingPlanAllocation {
+          sellingPlan {
+            id
+            name
+            recurringDeliveries
+          }
+          priceAdjustments {
+            perDeliveryPrice {
+              amount
+              currencyCode
+            }
+          }
+        }
+      }
+    `
+  }
+});
+```
+
+### Customizing `merchandise` data
+
+If the default `merchandise` data doesn't overlap with the needs of your project's cart, you can supply a `MERCHANDISE` fragment to the client's `customFragments`. If your project's cart UI is using product and variant data from Nacelle and only needs a handful of Shopify `Cart` properties (`barcode`, `currentlyNotInStock`, and `requiresShipping`), you could supply the following `MERCHANDISE` fragment to query only for the `merchandise` data on interest:
+
+```js
+const cartClient = createShopifyCartClient({
+  shopifyShopId: '<your-shop-id>',
+  shopifyStorefrontAccessToken: '<your-shopify-storefront-api-access-token>',
+  customFragments: {
+    MERCHANDISE: `
+      fragment ProductVariant_merchandise on ProductVariant {
+        barcode
+        currentlyNotInStock
+        requiresShipping
+      }
+    `
+  }
+});
+```
+
+<!-- LINKS -->
+
+[graphql-fragments-docs]: https://graphql.org/learn/queries/#fragments
+[nacelle-shopify-cart-fragments]: https://github.com/getnacelle/nacelle-js/tree/main/packages/shopify-cart/src/graphql/fragments
+[shopify-access-scopes]: https://shopify.dev/api/usage/access-scopes#unauthenticated-access-scopes
+[shopify-cart-object]: https://shopify.dev/api/storefront/2022-07/objects/Cart#top
+[shopify-cart-user-error]: https://shopify.dev/api/storefront/2022-07/objects/CartUserError
+[shopify-graphiql-explorer]: https://shopify.dev/custom-storefronts/tools/graphiql-storefront-api
```

Comment:
I think we can simplify this document structure a bit and give it a slightly better flow. 
```md
# Fragments in `@nacelle-shopify-cart`
<!-- Added the "flexibility of GQL fragments section here" --> 
If you'd like to query for additional fields on the [`Cart`][shopify-cart-object] object or its sub-properties, you can supply [GraphQL fragments][graphql-fragments-docs] when initializing the `cartClient`.

`@nacelle/shopify-cart` uses GraphQL queries and mutations to create and modify Shopify `Cart`s. We use GraphQL fragments to specify properties of interest for things like `cart.lines.merchandise`, `cart.discountAllocations`, and `userErrors`.

Because `Cart` data needs will vary project-to-project, it's important that developers have the power to specify their own properties of interest. By providing a way to supply `customFragments` to the `cartClient`, developers can enjoy the convenience of `@nacelle/shopify-cart` with the flexibility of GraphQL.

## Prerequisites
<!-- Prereq stuff goes here-->

## Initializing the `cartClient` with `customFragments`
<!-- Explain the different kinds of fragments in this section-->

## Default Fragments

## Extension Fragments 

## Replacement Fragments

## Examples
```

Reply:
Good idea! Implemented in https://github.com/getnacelle/nacelle-js/pull/229/commits/6599ea9c19402b152fa01ba7ffe6facb2988bd0e.